pydis_site/apps/api/tests/test_infractions.py

```diff
@@ -417,27 +424,37 @@ def test_unique_constraint_raises_integrity_error_on_second_active_of_same_type(
 
     def test_unique_constraint_accepts_active_infraction_after_inactive_infraction(self):
         """Do we accept an active infraction if the others of the same type are inactive?"""
-        Infraction.objects.create(
-            user=self.user,
-            actor=self.user,
-            type="ban",
-            active=False,
-            reason="The first inactive ban"
-        )
-        Infraction.objects.create(
-            user=self.user,
-            actor=self.user,
-            type="ban",
-            active=False,
-            reason="The second inactive ban"
-        )
-        Infraction.objects.create(
-            user=self.user,
-            actor=self.user,
-            type="ban",
-            active=True,
-            reason="The first active ban"
-        )
+        try:
+            Infraction.objects.create(
+                user=self.user,
+                actor=self.user,
+                type="ban",
+                active=False,
+                reason="The first inactive ban"
+            )
+            Infraction.objects.create(
+                user=self.user,
+                actor=self.user,
+                type="ban",
+                active=False,
+                reason="The second inactive ban"
+            )
+            Infraction.objects.create(
+                user=self.user,
+                actor=self.user,
+                type="ban",
+                active=True,
+                reason="The first active ban"
+            )
+        except IntegrityError:
+            self.fail("An unexpected IntegrityError was raised.")
+
+    @patch(f"{__name__}.Infraction")
+    def test_the_accepts_active_infraction_after_inactive_infractions_test(self, infraction_patch):
```

Comment:
This test name is confusing, perhaps unsurprisingly.

Reply:
I don't really feel like this test is necessary to be honest.

Reply:
It's necessary since we include tests in the coverage report and otherwise the `except`-block for the test we're testing here will never be run. 

We may be able to declare an exception to the coverage requirement for those specific lines, but I wrote to test to make sure it actually works like I intended it to, since it does have custom test logic.

Reply:
Oh right. I am not used to coverage being enabled for tests. I still think the name could be better by mentioning that it is specifically testing the exception being caught, but the name is already getting quite long 🤔 .